lib/interpret_and_optimize/helpers/pb_project.dart

```diff
@@ -25,4 +25,40 @@ class PBProject {
     _genProjectData = PBGenerationProjectData();
     _fileStructureStrategy = fileStructureStrategy;
   }
+
+  // Pass through the forest once checking all trees
+  // if two or more trees have the same name and same
+  // orientation, a `_n` counter is added at the end
+  // of their name
+  void cleanTreeParity() {
+    forest.sort((a, b) => a.rootNode.name.compareTo(b.rootNode.name));
+    var currentCounter = 0;
+    var nextCounter = currentCounter + 1;
+    var nameCounter = 1;
+
+    while (nextCounter < forest.length) {
+      var currentTree = forest[currentCounter];
+      var nextTree = forest[nextCounter];
+      if (currentTree.rootNode.name.compareTo(nextTree.rootNode.name) == 0) {
+        if (currentTree.data.orientation == nextTree.data.orientation) {
+          forest[nextCounter].rootNode.name =
+              forest[nextCounter].rootNode.name + '_$nameCounter';
+
+          nameCounter++;
+          nextCounter++;
+        } else {
+          currentCounter++;
```

Comment:
Rather than only increasing `currentCounter` by one, could we do something like this?
```
currentCounter = nextCounter;
nextCounter++;
```
In the case that there are no repeating names, currentCounter would simply increase by one. However, in the case that there are repeated names this would let us skip the trees we've already renamed rather than redundantly traversing them. 🤔 Let me know if you see flaws in this approach.

Reply:
You are right, this can work, so would this get rid of the `if` statements on line 51 & 57??
They look like the following:
``` 
if (currentCounter >= nextCounter) {
          nextCounter++;
        }
```

Reply:
Follow up in that, yup, they are no longer needed